lib/screens/qr_code_screen.dart

```diff
@@ -0,0 +1,105 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:qr_flutter/qr_flutter.dart';
+import 'package:superformula_qr_code/models/seed_model.dart';
+import 'package:superformula_qr_code/services/api_service.dart';
+import 'package:superformula_qr_code/shared/app_routes.dart';
+import 'package:http/http.dart' as http;
+
+class QrCodeScreen extends StatefulWidget {
+  const QrCodeScreen({Key? key}) : super(key: key);
+
+  @override
+  State<QrCodeScreen> createState() => _QrCodeScreenState();
+}
+
+class _QrCodeScreenState extends State<QrCodeScreen> {
+  // To avoid unecessary rebuilds, initialize api call & countdown stream outside of the build widget
+  final Future<SeedModel> _future = ApiService().getSeed(http.Client());
+
+  static const int _count = 15;
+  final Stream<int> _countdown = countdown(_count);
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('QR Code'),
+        centerTitle: true,
+      ),
+      body: StreamBuilder<dynamic>(
```

Comment:
I'm mostly familiar with Provider for state management, this is what we currently use at ForWhenI. I've used bloc in the past but we found it can be quite verbose for smaller applications.

Reply:
Sounds good. The project for this current opening heavily uses BLoC.

Reply:
Ah I see. Had I known that I would've made an effort to use BLoC for this assignment 😅.

With my current company we have a very small team. Do you guys use primarily use BLoC or just for this opening?

Reply:
Most of our teams use BLoC in some form right now (a decision made with the client(s)) for many different reasons. One of our projects was using Provider but has switched over to BLoC.

Reply:
Ah okay, interesting.

Well if you'd like me to, I could spend some time this week refactoring this to use BLoC. And I can send you a repo of an older personal project I made that used BLoC as well